controllers/release/adapter.go

```diff
@@ -478,3 +492,43 @@ func (a *Adapter) syncResources() error {
 
 	return a.syncer.SyncSnapshot(snapshot, releasePlanAdmission.Namespace)
 }
+
+// registerAttributionData updates the status of the Release being processed with the proper attribution author.
+func (a *Adapter) registerAttributionData(releasePlan *v1alpha1.ReleasePlan) error {
+	if a.release.IsAttributed() {
+		return nil
+	}
+
+	var author string
+	patch := client.MergeFrom(a.release.DeepCopy())
+
+	if a.release.Labels[metadata.AutomatedLabel] == "true" {
+		author = releasePlan.Labels[metadata.AuthorLabel]
+		if author == "" {
+			return fmt.Errorf("no author in the ReleasePlan found for automated release")
+		}
+		a.release.Status.Attribution.StandingAuthorization = true
+	} else {
+		author = a.release.Labels[metadata.AuthorLabel]
+		if author == "" { // this should be impossible
```

Comment:
This should be or it is actually impossible to match at the moment?  When I was writing tests there was cases in the adapter code (maybe they'd gone after dozen refactorings) that could not never happen, although no comments at all. So maybe this comment is also unnecessary.

Reply:
It should be impossible because webhooks prevent it from happening. However, in instances such as e2e or place where webhooks do not run, it could happen, so I figured it is probably best to check for it